pkg/sdn/plugin/node_iptables.go

```diff
@@ -91,7 +91,7 @@ func (n *NodeIPTables) syncIPTableRules() error {
 // Get openshift iptables rules
 func (n *NodeIPTables) getStaticNodeIPTablesRules() []FirewallRule {
 	return []FirewallRule{
-		{"nat", "POSTROUTING", []string{"-s", n.clusterNetworkCIDR, "!", "-d", n.clusterNetworkCIDR, "-j", "MASQUERADE"}},
+		{"nat", "POSTROUTING", []string{"-s", n.clusterNetworkCIDR, "-j", "MASQUERADE"}},
```

Comment:
Before, when we used docker for IPAM, docker's own iptables rules ended up applying to us, because we were configuring docker to use the same IP range as OpenShift. Since we're no longer reconfiguring docker's networking, we no longer get those rules, so we need to change our rules a bit to compensate.

In particular, we need a rule that says that if iptables sees traffic going from one pod to another pod, then it needs to masquerade it. That's because the only time that pod-to-pod traffic would end up being seen by iptables is if the traffic was originally pod-IP-to-service-IP, and then iptables DNATed the service IP to a pod IP. In that case we need to masquerade, because if we don't, then the destination pod would try to send its response directly back to the source pod, which would happen on the OVS bridge without iptables ever seeing it, and so the DNATing wouldn't get reversed, and so the source pod wouldn't recognize the packet. By masquerading it to the node's IP, we ensure that the service's response gets bounced out of OVS back to where iptables can see it again so it can un-DNAT it before passing it back to the source pod.

Since we already had a rule saying that pod-to-external traffic needs to be masqueraded, and we also need a rule saying that pod-to-pod traffic (that gets seen by iptables) needs to be masqueraded, we can just squash those rules together to get what you see here: any traffic from a pod that ends up being seen by iptables needs to be masqueraded.